docs/design/datamover.md

```diff
@@ -63,93 +63,85 @@ This design supports adding the data mover feature to the OADP operator and faci
-is used.
+`VolumeSnapshotBackup` CR supports a `volumesnapshotcontent` as the type of the backup object. Velero backup will wait for the `VolumeSnapshotBackup` to be complete. Once all the `VolumeSnapshotBackup` gets completed, Velero backup's status get updated accordingly.
 
 ```
-apiVersion: oadp.openshift.io/v1alpha1
-kind: DataMoverBackup
+apiVersion: datamover.oadp.openshift.io/v1alpha1
+kind: VolumeSnapshotBackup
 metadata:
-  name: <name>
-spec:
-  dataMoverClass: <DataMoverClass name> 
-  dataSourceRef:
-    apiGroup: <APIGroup>
-    kind: <PVC|VolumeSnapshotContent>
-    name: <name>
-  config:  //optional based on the datamover impl
-
-```
-### Data Mover Restore
-When a velero restore is triggered, the custom Velero CSI plugin looks for `DataMoverBackup` in the backup resources. If it encounters a `DataMoverBackup` resource, then the extended plugin (velero RestoreItemAction plugin) will create a `DataMoverRestore` CR in the app namespace. It will populate the CR with the details obtained from the `DataMoverBackup` resource. 
-
-```
-apiVersion: oadp.openshift.io/v1alpha1
-kind: DataMoverRestore
-metadata:
-  name: <name>
+  annotations:
+    datamover.io/restic-repository: <restic_repo>
+    datamover.io/source-pvc-name: <src_pvc_name>
+    datamover.io/source-pvc-size: <src_pvc_size>
+  labels:
+    velero.io/backup-name: <backup_name>
+  name: <vsb_name>
+  namespace: <ns>
 spec:
-  DataMoverClass: <DataMoverClass name>
-  destinationClaimRef:
-    name: <PVC_claim_name>
-    namespace: <namespace>
-  config:     //optional based on the datamover impl
-```
-Config section in the above CR is optional. It lets the user specify extra parameters needed by the data mover. For eg: VolSync data mover needs restic secret to perform backup & restore
-
-eg: 
-
-```
-apiVersion: oadp.openshift.io/v1alpha1
-kind: DataMoverRestore
+  protectedNamespace: <oadp_ns>
+  resticSecretRef:
+    name: <bsl_restic_secret>
+  volumeSnapshotContent:
+    name: <vsc_name>
+status:
+  phase: <vsb_status>
+  resticrepository: <restic_repo>
+  sourcePVCData:
+    name: <src_pvc_name>
+    size: <src_pvc_size>
+    storageClassName: <sc_name>
+  volumeSnapshotClassName: <volumesnapshotclass_name>
+```
+### Volume Snapshot Restore
+When a velero restore is triggered, the custom Velero CSI plugin looks for `VolumeSnapshotBackup` CR in the backup resources. If it encounters a `VolumeSnapshotBackup` resource, then the extended plugin (velero RestoreItemAction plugin) will create a `VolumeSnapshotRestore` CR in the app namespace. It will populate the CR with the details obtained from the `VolumeSnapshotBackup` resource. 
+
+Velero restore process will wait for the snapshot to be restored by `VolumeSnapshotRestore` controller and then proceeds with rest of the restore process.
```

Comment:
The velero CSI plugin waits for VSR to complete, which is after the VSR controller waits for VolSync replicationDestination to complete, before proceeding with the rest of restore.